codalab/rest/bundle.py

```diff
@@ -1,15 +1,546 @@
+
+    # Delete the actual bundle
+    if not dry_run:
+        if data_only:
+            # Just remove references to the data hashes
+            local.model.remove_data_hash_references(relevant_uuids)
+        else:
+            # Actually delete the bundle
+            local.model.delete_bundles(relevant_uuids)
+
+        # Update user statistics
+        local.model.update_user_disk_used(request.user.user_id)
+
+    # Delete the data_hash
+    for uuid in relevant_uuids_set:
+        # check first is needs to be deleted
+        bundle_location = local.bundle_store.get_bundle_location(uuid)
+        if os.path.lexists(bundle_location):
+            local.bundle_store.cleanup(uuid, dry_run)
+
+    # Return list of deleted ids
+    return json_api_meta({}, {'ids': relevant_uuids})
+
+
+@post('/bundle-permissions', apply=[AuthenticatedPlugin(), JsonApiPlugin()])
+def set_bundle_permissions():
+    many = isinstance(request.json['data'], list)
+    new_permissions = BundlePermissionSchema(
+        strict=True, many=many,
+    ).load(request.json).data
+
+    # Multiplex between single and bulk requests
+    if many:
+        set_bundle_permissions_helper(new_permissions)
+    else:
+        set_bundle_permissions_helper([new_permissions])
+
+    return BundlePermissionSchema(many=many).dump(new_permissions).data
+
+
+# TODO(sckoo): Return updated permissions
+def set_bundle_permissions_helper(new_permissions):
+    # Check permissions
+    bundle_uuids = [p['object_uuid'] for p in new_permissions]
+    check_bundles_have_all_permission(local.model, request.user, bundle_uuids)
+
+    # Multiplex between updating, adding, or deleting permissions
+    for p in new_permissions:
+        old_permission = local.model.get_group_bundle_permission(p['group_uuid'], p['object_uuid'])
+        new_permission = p['permission']
+        if new_permission > 0:
+            if old_permission > 0:
+                local.model.update_bundle_permission(p['group_uuid'], p['object_uuid'], new_permission)
+            else:
+                local.model.add_bundle_permission(p['group_uuid'], p['object_uuid'], new_permission)
+        else:
+            if old_permission > 0:
+                local.model.delete_bundle_permission(p['group_uuid'], p['object_uuid'])
+
+
+@get('/bundles/<uuid:re:%s>/contents/info/' % spec_util.UUID_STR, apply=AuthenticatedPlugin())
+@get('/bundles/<uuid:re:%s>/contents/info/<path:path>' % spec_util.UUID_STR, apply=AuthenticatedPlugin())
+def fetch_contents_info(uuid, path=''):
+    depth = query_get_type(int, 'depth', default=0)
+    if depth < 0:
+        abort(httplib.BAD_REQUEST, "Depth must be at least 0")
+
+    check_bundles_have_read_permission(local.model, request.user, [uuid])
+    return {
+        'data': local.download_manager.get_target_info(uuid, path, depth)
+    }
+
+
+@get('/bundles/<uuid:re:%s>/contents/blob/' % spec_util.UUID_STR, apply=AuthenticatedPlugin())
```

Comment:
Hm yeah that's why I asked you about this yesterday. What are your thoughts? We could 1) keep this as the singular "bundle" and just deal with it being different from the other, 2) change all the bundle methods to be singular, 3) change ALL the methods to be singular, 4) rename this to plural as in the current PR.